docs/design/messaging/message-round-trip.puml

```diff
@@ -0,0 +1,52 @@
+@startuml
+title message round trip
+
+collections handler
+collections service
+participant application
+participant transport
+participant remote
+collections remote_handler
+
+activate handler
+handler -> handler : produce message payload\nand message cookie\nwith unique request id\n[payload, cookie(0)]
+handler -> service : [payload, cookie(0)]
```

Comment:
nit: cookie[0], cookie[1], .... cookie[n] as for me

Reply:
ok...

Reply:
it's on lines bellow as well